app/code/community/Razorpay/Payments/Model/Paymentmethod.php

```diff
@@ -230,4 +227,81 @@ public function getConfigData($field, $storeId = null)
 
         return Mage::getStoreConfig($path, $storeId);
     }
+
+    /**
+     * We need to use this method to load all the razorpay-php files
+     */
+    public function requireAllRazorpayFiles()
+    {
+        $baseDir = Mage::getBaseDir('lib') . DS . 'razorpay-php';
+
+        $apiClassesBase = $baseDir . DS . 'src';
+        $errorClassesBase = $baseDir . DS . 'src/Errors';
+        $requestClassesBase = $baseDir . DS . 'libs/Requests-1.6.1/library/Requests';
+
+        // Require requests class
+        require_once $requestClassesBase . '.php';
+
+        // Require all requests files first
+        $this->recursiveReadDirectory($requestClassesBase);
+
+        // Require Entity, Resource and ArrayableInterface first
+        require_once $apiClassesBase . DS . 'Resource.php';
+        require_once $apiClassesBase . DS . 'ArrayableInterface.php';
+        require_once $apiClassesBase . DS . 'Entity.php';
+
+        // Requiring base error class first
+        require_once $errorClassesBase . DS . 'Error.php';
+
+        // Require all src files first
+        foreach (scandir($apiClassesBase) as $file)
```

Comment:
`scandir` doesn't preserve order, and this is a terrible way to do this. Is there nothing that magento provides that preserves order and loads them for us?